homeassistant/components/wemo.py

```diff
@@ -97,7 +99,7 @@ def discovery_dispatch(service, discovery_info):
             device = pywemo.discovery.device_from_description(url, None)
 
         discovery_info = {
-            'model_name': device.model_name,
+            'model_name': re.search(r"(?:\S+\s+)(\S+)", repr(device)).group(1),
```

Comment:
We shouldn't use repr. Instead we should get the data where repr is getting it from. Regexing repr is bound to break in the future as repr is not at all guaranteed API stability.

Reply:
As far as I can tell repr is where the information comes from: https://github.com/pavoni/pywemo/blob/master/pywemo/ouimeaux_device/motion.py

Reply:
I still think this should be fixed in pywemo, not here. Ie pywemo should make sure that `device.model_name` is correct and makes sense.